### PR TITLE
[all hosts] (Visual Studio) Update JS->TS instructions to not rely on the jQuery nuget package

### PR DESCRIPTION
**Pull request by AlexJerabek**
The current article has a step to add the jquery.TypeScript.DefinitelyTyped nuget package. That doesn't appear to exist anymore. This PR retools the instructions to still get the right type definitions for IntelliSense into the project.

I'd **really** appreciate someone verify the instructions work. I tried to record the minimum steps required but might have missed something.

### CODE REVIEW

File: docs/develop/convert-javascript-to-typescript.md [hunk 1]
```diff
@@ -70,6 +72,26 @@ You can use the Office Add-in template in Visual Studio to create an add-in that
 
 1. Save the file. For more information on **tsconfig.json** settings, see [What is a tsconfig.json?](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html)
 
+### Create an npm configuration file
+
+1. In **Solution Explorer**, right-click (or select and hold) the ASP.NET web project and choose **Add > New Item**. The ASP.NET web project has your project name with the text `Web` appended to the end.
+1. In the **Add New Item** dialog, select **npm Configuration File** to create a **package.json** file and then choose **Add**.
```
Comment (Rick-Kirkham) [inline docs/develop/convert-javascript-to-typescript.md:78]: I don't have an option **npm Configuration File** on that dialog. I'm on Visual Studio Enterprise 2022 (64-bit) 
Version 17.10.4

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]
I'm blocked due to a UI discrepancy.

File: docs/develop/convert-javascript-to-typescript.md [hunk 2]
```diff
@@ -70,6 +72,26 @@ You can use the Office Add-in template in Visual Studio to create an add-in that
...
+      "name": "asp.net",
+      "private": true,
+      "devDependencies": {
+        "@types/jquery": "^3.5.30"
+      }
+    }
+    ```
+
+1. Save the file.
+1. Select the ASP.NET web project has your project name with the text `Web` appended to the end. Under the **npm** section, set both **Restore On Project Open** and **Restore On Save** to "True".
```
Comment (Rick-Kirkham) [inline docs/develop/convert-javascript-to-typescript.md:93]:
```suggestion
1. Select the ASP.NET web project that has your project name with the text `Web` appended to the end. Under the **npm** section, set both **Restore On Project Open** and **Restore On Save** to "True".
```

File: docs/develop/convert-javascript-to-typescript.md (hunk 1 above)
Comment (AlexJerabek) [reply docs/develop/convert-javascript-to-typescript.md:78]: I'm on the same version, so I think the problem is in my instructions. 

The **Add New Item** step has to be done to this part of the project:
image

Once there, you should have this option:
image

If you try to add a new item from the project root, you don't get those options.
image

The `npm install` step should be done when you set those npm settings to true (at least, I never ran `npm install` on this project).
image

This is the article I heavily relied on for this: https://learn.microsoft.com/en-us/visualstudio/javascript/npm-package-management?view=vs-2022#aspnet-core-projects

If this information doesn't help, I'll try again.

Comment (Rick-Kirkham) [reply docs/develop/convert-javascript-to-typescript.md:78]: OK. So the missing step is that in the **Add New Item** dialog, the user has to search for "npm".

File: docs/develop/convert-javascript-to-typescript.md [hunk 3]
```diff
@@ -33,11 +36,10 @@ You can use the Office Add-in template in Visual Studio to create an add-in that
 
 ## Convert the add-in project to TypeScript
 
-### Add Nuget packages
+### Add Nuget package
 
 1. Open the Nuget package manager by choosing **Tools > Nuget Package Manager > Manage Nuget Packages for Solution**
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/convert-javascript-to-typescript.md:41]: Each item should be bolded individually. Here and other instances

```suggestion
1. Open the Nuget package manager by choosing **Tools** > **Nuget Package Manager** > **Manage Nuget Packages for Solution**
```

File: docs/develop/convert-javascript-to-typescript.md [hunk 4]
```diff
@@ -33,19 +36,18 @@ You can use the Office Add-in template in Visual Studio to create an add-in that
...
+1. Open the Nuget package manager by choosing **Tools** > **Nuget Package Manager** > **Manage Nuget Packages for Solution**
+1. Select the **Browse** tab. Search for and select **Microsoft.TypeScript.MSBuild**. Install this package to the ASP.NET web project, or update it if it's already installed. The ASP.NET web project has your project name with the text `Web` appended to the end. This will ensure the project will transpile to JavaScript when the build runs.
 
 > [!NOTE]
 > In your TypeScript project, you can have a mix of TypeScript and JavaScript files and your project will compile. This is because TypeScript is a typed superset of JavaScript that compiles JavaScript.
 
 ### Create a TypeScript config file
 
 1. In **Solution Explorer**, right-click (or select and hold) the ASP.NET web project and choose **Add > New Item**. The ASP.NET web project has your project name with the text `Web` appended to the end.
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/convert-javascript-to-typescript.md:49]:
```suggestion
1. In **Solution Explorer**, right-click (or select and hold) the ASP.NET web project and choose **Add** > **New Item**. The ASP.NET web project has your project name with the text `Web` appended to the end.
```

File: docs/develop/convert-javascript-to-typescript.md [hunk 5]
```diff
@@ -70,6 +72,26 @@ You can use the Office Add-in template in Visual Studio to create an add-in that
 
 1. Save the file. For more information on **tsconfig.json** settings, see [What is a tsconfig.json?](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html)
 
+### Create an npm configuration file
+
+1. In **Solution Explorer**, right-click (or select and hold) the ASP.NET web project and choose **Add > New Item**. The ASP.NET web project has your project name with the text `Web` appended to the end.
```
Comment (ElizabethSamuel-MSFT) [inline docs/develop/convert-javascript-to-typescript.md:77]: Remember to change these and any others

```suggestion
1. In **Solution Explorer**, right-click (or select and hold) the ASP.NET web project and choose **Add** > **New Item**. The ASP.NET web project has your project name with the text `Web` appended to the end.
```

**Review by Rick-Kirkham — APPROVED** [review]

File: docs/develop/convert-javascript-to-typescript.md (hunk 1 above)
Comment (Rick-Kirkham) [reply docs/develop/convert-javascript-to-typescript.md]: @AlexJerabek Restarted VS this morning and without making any changes, the project ran fine. A theory: The installation of the jQuery types is asynchronous and it just hadn't completed when I ran the project yesterday. Alter

**Pull request merged**